### DISCUSSION TITLE
Real time progress and logging

### DISCUSSION THREAD

**Discussion by jamesst20** [bensheldon/good_job#1386] [Ideas]
Similar to https://github.com/kenaniah/sidekiq-status

You can set your Job progress and it is displayed like this

![Capture d’écran, le 2024-07-02 à 10 05 34](https://github.com/bensheldon/good_job/assets/490234/3cc86a8f-7e49-4c41-87c6-564b14f14915)

You can also display any kind of information so let's say your Job is syncing 1000 items, you can log stuffs like

```
1/1000: Already up to date
2/1000: Pushed to the API
....
```

![Capture d’écran, le 2024-07-02 à 10 08 26](https://github.com/bensheldon/good_job/assets/490234/9d123658-1e43-4186-8d30-e61802a911b0)

**Comment by mbajur** [comment]
That would be awesome to have it and it seems to be relatively easy to implement. I would personally even prefer the `store/retrieve` API in order to be able to save some sort of artifact data to investigate after job completed but that would probably be part of the same API :) Progress part might be accomplished with Batches towards which i usually gravitate to every time i need to track progress.

I think it might be implemented as a plugin, not part of the core (and merged into core eventually, if author is interested in that). I might play around with that in upcoming days.

**Reply by bensheldon** [reply]
I'm supportive if you wanted to try to add something like this.

I wonder if the value could be inserted directly into the serialized_params rather than requiring a new database column. As long as it's not being queried/filtered by on the dashboard (aka display only) that could be sufficient

**Reply by mbajur** [reply]
Wouldn't that be a bit unexpected to keep variables defined/assigned during runtime in the column which name suggests these are input attributes and column which is indeed used exclusively for that? Just thinking...I can imagine getting confused by this 🤔 

On one hand, obviously simplicity which is great. But on the other hand one might end up with massive amounts of data assigned on runtime in the column which also holds input params which might be confusing
